### ISSUE TITLE
Bug in `dht2` with `effort_sum`

### ISSUE THREAD

**Issue by erex** [DistanceDevelopment/Distance#41]
Student in distance workshop found using uniform cosine detection function with the sika deer data throws an error when `dht2` is called.  Here is reproducible example.  The fitted detection function looks fine, but I don't know what hassle the uniform cosine might cause in `dht2`

```
library(Distance)
data(sikadeer)
conversion.factor <- convert_units("centimeter", "kilometer", "square kilometer")
deer.df <- ds(sikadeer, key="unif", adj="cos",truncation="10%")
mult <- list(creation = data.frame(rate=25, SE=0),
              decay    = data.frame(rate=163, SE=14))
# note this is the same call to dht2 that works when a half normal is detection function
deer_ests <- dht2(deer.df, flatfile=sikadeer, strat_formula=~Region.Label,
                   convert_units=conversion.factor, multipliers=mult, 
                   stratification="effort_sum",total_area=13.9)
```

**Error in x[!dup, ] : (subscript) logical subscript too long**

**Comment by dill** [comment]
Frustratingly (though perhaps inevitably, given new mrds was accepted at 
CRAN yesterday) this is an mrds bug (which only affects predict() on new 
data when a uniform key is used). I have patched mrds in 822e122.

**Comment by erex** [comment]
glad you found it; of course it would be buried inside mrds...
